src/PostgREST/App.hs

```diff
@@ -101,37 +102,31 @@ app dbStructure conf apiRequest =
   case (iAction apiRequest, iTarget apiRequest, iPayload apiRequest) of
 
     (ActionRead, TargetIdent qi, Nothing) ->
-      serves [CTApplicationJSON, CTTextCSV] (iAccepts apiRequest) $ \contentType ->
-      case readSqlParts of
-        Left e -> return $ responseLBS status400 [jsonH] $ toS e
-        Right (q, cq) -> do
-          let singular = iPreferSingular apiRequest
-              stm = createReadStatement q cq singular
-                    shouldCount (contentType == CTTextCSV)
-          respondToRange $ do
+      servesMatchingContentTypes' $ \contentType ->
```

Comment:
The tick mark naming convention confused me. Maybe it's worthwhile to use the non-tick version and pass in the arguments repetitively.